### PR TITLE
Removed config from agent controller

### PR DESCRIPTION
**Pull request by neubig**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**

This PR removes global config from the agent controller.

**Give a summary of what the PR does, explaining any non-trivial design decisions**

There is one tricky part about this, which is that when the agent delegates it needs to grab the LLMConfig for the agent that is delegated to. I considered two options:

1. Pass in `AppConfig` to `AgentController`
2. Pass in an `agent_to_llm_config` map that only contains the dictionary mapping from the agent class string to the appropriate `LLMConfig`.

Eventually I opted for the second option, because it makes it more clear exactly what information `AgentController` uses instead of throwing in the entire config, which obscures this.

**Other references**
Part of #2758

### CODE REVIEW

**Review by iFurySt — APPROVED** [review]
LGTM, thanks!

**Comment by neubig** [comment]
Hmm, frontend tests are failing here and I'm not sure why because I didn't change anything in the frontend... @amanape do you have any idea why this might be broken? https://github.com/OpenDevin/OpenDevin/actions/runs/10020070149/job/27697328000?pr=3038

**Comment by amanape** [comment]
@neubig I have re-run the job and it passed. I have noticed that the frontend tests have become a little flakey, and many of the "failing" tests occur with the nextui select component. It's odd but I have encountered difficulties testing those components early when I first started setting up tests.

**Comment by neubig** [comment]
Ah, got it, thanks! I reran once but got the same error, so I was confused

File: tests/integration/test_agent.py [hunk 1]
```diff
@@ -21,32 +21,19 @@
...
     assert final_state is not None
     assert final_state.agent_state == AgentState.STOPPED
     assert final_state.last_error is None
-    # number of LLM conversations should be the same as number of prompt/response
-    # log files under mock/[agent]/[test_name] folder. If not, it means there are
-    # redundant prompt/response log files checked into the repository.
-    num_of_conversations = get_number_of_prompts(test_name)
-    assert num_of_conversations > 0
-    # we mock the cost of every conversation to be 1 USD
-    assert final_state.metrics.accumulated_cost == num_of_conversations
```
Comment (li-boxuan) [inline tests/integration/test_agent.py:49]: This removal seems to due to problematic merge conflict resolution

Comment (neubig) [reply tests/integration/test_agent.py]: Woops, sorry, thanks for the catch!

File: tests/integration/mock/CodeActAgent/test_browse_internet/prompt_001.log [hunk 2]
```diff
@@ -401,4 +401,4 @@ NOW, LET'S START!
 
 Browse localhost:8000, and tell me the ultimate answer to life. Do not ask me for confirmation at any point.
 
-ENVIRONMENT REMINDER: You have 14 turns left to complete the task. When finished reply with <finish></finish>
+ENVIRONMENT REMINDER: You have 49 turns left to complete the task. When finished reply with <finish></finish>
```
Comment (li-boxuan) [inline tests/integration/mock/CodeActAgent/test_browse_internet/prompt_001.log:404]: This seems problematic. It looks like it's using max_iterations = 50, presumably from your `config.toml`. 

In `regenerate.sh` we defined it as `MAX_ITERATIONS=15`

Comment (neubig) [reply tests/integration/mock/CodeActAgent/test_browse_internet/prompt_001.log]: Sorry, this is fixed!

**Pull request merged**